script/vsts/upload-artifacts.js

```diff
@@ -65,27 +65,47 @@ async function uploadArtifacts() {
     return;
   }
 
-  console.log(
-    `Uploading ${
-      assets.length
-    } release assets for ${releaseVersion} to S3 under '${bucketPath}'`
-  );
-
-  await uploadToS3(
-    process.env.ATOM_RELEASES_S3_KEY,
-    process.env.ATOM_RELEASES_S3_SECRET,
-    process.env.ATOM_RELEASES_S3_BUCKET,
-    bucketPath,
-    assets
-  );
+  if (
+    process.env.ATOM_RELEASES_S3_KEY &&
+    process.env.ATOM_RELEASES_S3_KEY !== '$(ATOM_RELEASES_S3_KEY)' &&
+    process.env.ATOM_RELEASES_S3_SECRET &&
+    process.env.ATOM_RELEASES_S3_SECRET !== '$(ATOM_RELEASES_S3_SECRET)'
```

Comment:
This check is redundant.
```suggestion
    process.env.ATOM_RELEASES_S3_KEY &&
    process.env.ATOM_RELEASES_S3_SECRET
```

Reply:
The one we need to have is `process.env.ATOM_RELEASES_S3_SECRET !== '$(ATOM_RELEASES_S3_SECRET)'`.

Unless we want to explicitly set ALL of these vars to be empty.

---

They are passed to the script with `env:` blocks in the yaml.

https://github.com/atom-ide-community/atom/blob/cfd4a0b2ce0977acc493517a841aa55f2ebb77a1/script/vsts/release-branch-build.yml#L54-L60

Anything passed via an `env:` block has one of the following states:
- Holds the actual value it was set to
  - (when set in Azure DevOps UI or pipeline yaml)
- holds the value `$(VAR_NAME_HERE)`, as a string
  - (when unset in Azure DevOps UI or pipeline yml)

It will not be empty. (unless we explicitly set it empty in Azure DevOps UI, which I don't want to do). [docs (see chart row for macro syntax).](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#runtime-expression-syntax)

---

As these values are sensitive/private credentials, they are stored as secret variables on the Azure DevOps side of things, they  _must_ be passed in an `env:` block, or else scripts will not have access to the value. This is a security feature to ensure only secret variables explicitly passed are available in scripts. [docs.](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#secret-variables)

> Unlike a normal variable, they are not automatically decrypted into environment variables for scripts. You need to explicitly map secret variables.

Reply:
Checking if an env var is truthy `if(process.env.ATOM_RELEASES_S3_KEY)` is the redundant check.

(`process.env.ATOM_RELEASES_S3_KEY &&` could be deleted, not the other one).

I want to keep this second check in case we ever want to use GitHub Actions or something. If you are 100% adamant this must be removed then I guess I will, since this is strictly CI, and we only run in Azure DevOps right now.

I prefer to leave it in.